Rewrite some sections of the documentation to match what we deliver on
OI, or remove sections that are of no relevance to OI.

--- mercurial-5.1.1/hgext/hgk.py
+++ mercurial-5.1.1/hgext/hgk.py
@@ -13,13 +13,12 @@ distributed with Mercurial.)
 
 hgk consists of two parts: a Tcl script that does the displaying and
 querying of information, and an extension to Mercurial named hgk.py,
-which provides hooks for hgk to get information. hgk can be found in
-the contrib directory, and the extension is shipped in the hgext
-repository, and needs to be enabled.
+which provides hooks for hgk to get information.
 
-The :hg:`view` command will launch the hgk Tcl script. For this command
-to work, hgk must be in your search path. Alternately, you can specify
-the path to hgk in your configuration file::
+The :hg:`view` command will launch the hgk Tcl script. The script is
+shipped in /usr/demo/mercurial, and hgk needs no configuration to find
+it.  You can specify the path to an alternate hgk in your configuration
+file::
 
   [hgk]
   path = /location/of/hgk
--- mercurial-5.9.3/mercurial/helptext/config.txt-orig	Tue Oct 26 08:23:49 2021
+++ mercurial-5.9.3/mercurial/helptext/config.txt	Sat Oct 30 16:03:19 2021
@@ -36,10 +36,6 @@
 
 Global configuration like the username setting is typically put into:
 
-.. container:: windows
-
-  - ``%USERPROFILE%\mercurial.ini`` (on Windows)
-
 .. container:: unix.plan9
 
   - ``$HOME/.hgrc`` (on Unix, Plan9)
@@ -52,7 +48,7 @@
 
 .. container:: verbose.unix
 
-  On Unix, the following files are consulted:
+  The following files are consulted:
 
   - ``<repo>/.hg/hgrc-not-shared`` (per-repository)
   - ``<repo>/.hg/hgrc`` (per-repository)
@@ -64,42 +60,6 @@
   - ``/etc/mercurial/hgrc.d/*.rc`` (per-system)
   - ``<internal>/*.rc`` (defaults)
 
-.. container:: verbose.windows
-
-  On Windows, the following files are consulted:
-
-  - ``<repo>/.hg/hgrc-not-shared`` (per-repository)
-  - ``<repo>/.hg/hgrc`` (per-repository)
-  - ``%USERPROFILE%\.hgrc`` (per-user)
-  - ``%USERPROFILE%\Mercurial.ini`` (per-user)
-  - ``%HOME%\.hgrc`` (per-user)
-  - ``%HOME%\Mercurial.ini`` (per-user)
-  - ``HKEY_LOCAL_MACHINE\SOFTWARE\Mercurial`` (per-system)
-  - ``<install-dir>\hgrc.d\*.rc`` (per-installation)
-  - ``<install-dir>\Mercurial.ini`` (per-installation)
-  - ``%PROGRAMDATA%\Mercurial\hgrc`` (per-system)
-  - ``%PROGRAMDATA%\Mercurial\Mercurial.ini`` (per-system)
-  - ``%PROGRAMDATA%\Mercurial\hgrc.d\*.rc`` (per-system)
-  - ``<internal>/*.rc`` (defaults)
-
-  .. note::
-
-   The registry key ``HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Mercurial``
-   is used when running 32-bit Python on 64-bit Windows.
-
-.. container:: verbose.plan9
-
-  On Plan9, the following files are consulted:
-
-  - ``<repo>/.hg/hgrc-not-shared`` (per-repository)
-  - ``<repo>/.hg/hgrc`` (per-repository)
-  - ``$home/lib/hgrc`` (per-user)
-  - ``<install-root>/lib/mercurial/hgrc`` (per-installation)
-  - ``<install-root>/lib/mercurial/hgrc.d/*.rc`` (per-installation)
-  - ``/lib/mercurial/hgrc`` (per-system)
-  - ``/lib/mercurial/hgrc.d/*.rc`` (per-system)
-  - ``<internal>/*.rc`` (defaults)
-
 Per-repository configuration options only apply in a
 particular repository. This file is not version-controlled, and
 will not get transferred during a "clone" operation. Options in
@@ -107,7 +67,7 @@
 
 .. container:: unix.plan9
 
-  On Plan 9 and Unix, most of this file will be ignored if it doesn't
+  On Unix, most of this file will be ignored if it doesn't
   belong to a trusted user or to a trusted group. See
   :hg:`help config.trusted` for more details.
 
@@ -129,10 +89,8 @@
 
 Per-installation configuration files are for the system on
 which Mercurial is running. Options in these files apply to all
-Mercurial commands executed by any user in any directory. Registry
-keys contain PATH-like strings, every part of which must reference
-a ``Mercurial.ini`` file or be a directory where ``*.rc`` files will
-be read.  Mercurial checks each of these locations in the specified
+Mercurial commands executed by any user in any directory.
+Mercurial checks each of these locations in the specified
 order until one or more configuration files are detected.
 
 Per-system configuration files are for the system on which Mercurial
@@ -699,14 +657,6 @@
 of an empty temporary file, where the filtered data must be written by
 the command.
 
-.. container:: windows
-
-   .. note::
-
-     The tempfile mechanism is recommended for Windows systems,
-     where the standard shell I/O redirection operators often have
-     strange effects and may corrupt the contents of your files.
-
 This filter mechanism is used internally by the ``eol`` extension to
 translate line ending characters between Windows (CRLF) and Unix (LF)
 format. We suggest you use the ``eol`` extension for convenience.
@@ -1052,23 +1002,6 @@
 of the hook in the config, respectively. In the example above, this will
 be ``$HG_HOOKTYPE=incoming`` and ``$HG_HOOKNAME=incoming.email``.
 
-.. container:: windows
-
-  Some basic Unix syntax can be enabled for portability, including ``$VAR``
-  and ``${VAR}`` style variables.  A ``~`` followed by ``\`` or ``/`` will
-  be expanded to ``%USERPROFILE%`` to simulate a subset of tilde expansion
-  on Unix.  To use a literal ``$`` or ``~``, it must be escaped with a back
-  slash or inside of a strong quote.  Strong quotes will be replaced by
-  double quotes after processing.
-
-  This feature is enabled by adding a prefix of ``tonative.`` to the hook
-  name on a new line, and setting it to ``True``.  For example::
-
-    [hooks]
-    incoming.autobuild = /my/build/hook
-    # enable translation to cmd.exe syntax for autobuild hook
-    tonative.incoming.autobuild = True
-
 ``changegroup``
   Run after a changegroup has been added via push, pull or unbundle.  The ID of
   the first new changeset is in ``$HG_NODE`` and last is in ``$HG_NODE_LAST``.
@@ -1278,13 +1211,6 @@
    Also, hooks like "commit" will be called in all contexts that
    generate a commit (e.g. tag) and not just the commit command.
 
-.. note::
-
-   Environment variables with empty values may not be passed to
-   hooks on platforms such as Windows. As an example, ``$HG_PARENT2``
-   will have an empty value under Unix-like platforms for non-merge
-   changesets, while it will not be available at all under Windows.
-
 The syntax for Python hooks is as follows::
 
   hookname = python:modulename.submodule.callable
@@ -1543,7 +1469,6 @@
 
   # Define new tool
   myHtmlTool.args = -m $local $other $base $output
-  myHtmlTool.regkey = Software\FooSoftware\HtmlMerge
   myHtmlTool.priority = 1
 
 Supported arguments:
@@ -1555,11 +1480,6 @@
 ``executable``
   Either just the name of the executable or its pathname.
 
-  .. container:: windows
-
-    On Windows, the path can use environment variables with ${ProgramFiles}
-    syntax.
-
   (default: the tool name)
 
 ``args``
@@ -1628,30 +1548,6 @@
   if ``premerge`` is ``keep` or ``keep-merge3``. See the corresponding variable
   in ``[ui]`` for more information.
 
-.. container:: windows
-
-  ``regkey``
-    Windows registry key which describes install location of this
-    tool. Mercurial will search for this key first under
-    ``HKEY_CURRENT_USER`` and then under ``HKEY_LOCAL_MACHINE``.
-    (default: None)
-
-  ``regkeyalt``
-    An alternate Windows registry key to try if the first key is not
-    found.  The alternate key uses the same ``regname`` and ``regappend``
-    semantics of the primary key.  The most common use for this key
-    is to search for 32bit applications on 64bit operating systems.
-    (default: None)
-
-  ``regname``
-    Name of value to read from specified registry key.
-    (default: the unnamed (default) value)
-
-  ``regappend``
-    String to append to the value read from the registry, typically
-    the executable name of the tool.
-    (default: None)
-
 ``pager``
 ---------
 
@@ -2610,10 +2506,6 @@
     ``false``
       Alias for ``ignore``.
 
-    .. container:: windows
-
-      On Windows, this configuration option is ignored and the command aborted.
-
 ``pre-merge-tool-output-template``
     (DEPRECATED) Use ``command-template.pre-merge-tool-output`` instead.
 
--- mercurial-5.1.1/mercurial/helptext/hg.1.txt
+++ mercurial-5.1.1/mercurial/helptext/hg.1.txt
@@ -85,6 +85,11 @@ This file can be used to define local ta
     repositories. The file format is the same as for ``.hgtags``, but it is
     encoded using the local system encoding.
 
+``/usr/demo/mercurial``
+    This directory contains assorted files which are part of the Mercurial
+    distribution, but not core to its functionality.  They will generally
+    need to be copied elsewhere to be of use.
+
 Some commands (e.g. revert) produce backup files ending in ``.orig``,
 if the ``.orig`` file already exists and is not tracked by Mercurial,
 it will be overwritten.
